src/allmydata/scripts/create_node.py

```diff
@@ -85,11 +130,30 @@ def write_node_config(c, config):
         webport = ""
     c.write("web.port = %s\n" % (webport.encode('utf-8'),))
     c.write("web.static = public_html\n")
-    c.write("# to prevent the Tub from listening at all, use this:\n")
-    c.write("#  tub.port = disabled\n")
-    c.write("#  tub.location = disabled\n")
-    c.write("#tub.port =\n")
-    c.write("#tub.location = \n")
+
+    if 'hostname' in config and config['hostname'] is not None:
+        print "HOSTNAME"
+        new_port = iputil.allocate_tcp_port()
+        c.write("tub.port = tcp:%s\n" % new_port)
+        c.write("tub.location = tcp:%s:%s\n" % (config.get('hostname').encode('utf-8'), new_port))
+    elif 'listen' in config and config['listen'] == "tor":
+        raise NotImplementedError("This feature addition is being tracked by this ticket:" +
+                                  "https://tahoe-lafs.org/trac/tahoe-lafs/ticket/2490")
+    elif 'listen' in config and config['listen'] == "i2p":
+        raise NotImplementedError("This feature addition is being tracked by this ticket:" +
+                                  "https://tahoe-lafs.org/trac/tahoe-lafs/ticket/2490")
```

Comment:
If ``config['listen']`` is set to something that we don't recognize at all, that should be an error.